common/src/main/java/org/jboss/gm/common/logging/FilteringCustomLogger.java

```diff
@@ -42,13 +51,12 @@ private FilteringCustomLogger(OutputEventListener outputEventListener) {
     public void onOutput(OutputEvent event) {
         LogEvent logEvent = (LogEvent) event;
         if (ignoreCategories.stream().noneMatch(i -> logEvent.getCategory().startsWith(i))) {
-            if (acceptableCategories.stream().noneMatch(i -> logEvent.getCategory().startsWith(i))) {
-                // Can't use logger to output a warning as causes stack overflow.
-                // TODO: Remove?
-                // System.err.println("Unknown event using category " + logEvent.getCategory() + " : ");
+            if (logEvent.getCategory().startsWith("org.commonjava.maven.ext")) {
+                delegate.onOutput(new LogEvent(logEvent.getTimestamp(), logEvent.getCategory(), LogLevel.LIFECYCLE,
+                        logEvent.getMessage(), logEvent.getThrowable()));
+            } else {
                 delegate.onOutput(event);
             }
-            delegate.onOutput(event);
```

Comment:
It looks like there was fall-through here which would make it print twice (on line 49 and again on line 51 if the expression evaluates to `true`) and was meant to have an `else` there.

Reply:
Yes, historically it had a different code path within the block, but was changed and wasn't updated.